torchvision/utils.py

```diff
@@ -199,10 +199,16 @@ def draw_bounding_boxes(
 
     txt_font = ImageFont.load_default() if font is None else ImageFont.truetype(font=font, size=font_size)
 
+    if colors is None:
+        if labels is None:
+            colors = _generate_color_palette(len(img_boxes))
+        else:
+            assert len(labels) == len(img_boxes)
+            label_color_map = dict(zip(labels, _generate_color_palette(len(labels))))
```

Comment:
Since now we agree that len(labels) == len(img_boxes) Line 207 and 204 can be simplified.